src/ConnectedMode/Migration/IConnectedModeMigration.cs

```diff
@@ -0,0 +1,64 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.Integration;
+
+namespace SonarLint.VisualStudio.ConnectedMode.Migration
+{
+    /// <summary>
+    /// Handle the non-UI part of migrating to the new unintrusive Connected Mode
+    /// file settings
+    /// </summary>
+    internal interface IConnectedModeMigration
+    {
+        // TODO: decide if the method should return a list of specific manual cleanup
+        // steps that the user still needs to perform
+        Task MigrateAsync(IProgress<MigrationProgress> progress, CancellationToken token);
```

Comment:
I've added a simple interface and basic implementation as a starting point.
I expect the API to change as we work out what we really need, but this will allow us to work on the UI and back end separately.